### PR TITLE
store: send Snap-Device-Location header with cloud information

### PR DESCRIPTION
**Pull request by wgrant**
The Store currently gets network location hints from a device using the download API's Snap-CDN header. But it is useful for Store operations to have that information even in e.g. refresh, so send the same cloud information to all APIs in a new header.

The Store might eventually use that if Snap-CDN isn't present, leaving Snap-CDN to handle only the "none" case, but for now leave that alone for compatibility.

### CODE REVIEW

**Review by mardy — APPROVED** [review]
LGTM, thanks!

But please wait for Samuele's review before merging.

File: store/store_download.go
Comment (pedronis) [inline store/store_download.go:171]: now download requests will have both Snap-CDN and Snap-Device-Location, I don't know if we care presently to phase the older one out?

**Review by pedronis — COMMENTED** [review]
lgtm, one consideration

File: store/store_download.go
Comment (wgrant) [reply store/store_download.go:171]: As I mentioned in the commit message, I think we will probably end up leaving `Snap-CDN` for only the `Snap-CDN: none` case. Once the new snapd is out in the wild I'll make the store use `Snap-Device-Location` if `Snap-CDN` isn't present.

Comment (pedronis) [reply store/store_download.go:171]: sorry, it seems a skipped reading the summary closely when I reviewed this

**Review by pedronis — APPROVED** [review]
thanks

File: store/store.go
Comment (Meulengracht) [inline store/store.go:748]: Nitpick: I know you just moved the code, but it seems it would make sense to reverse the if check here:
```
if cloudInfo == nil {
    return "", nil 
}
...
```

**Review by Meulengracht — APPROVED** [review]
Thanks! Just a small nitpick :-)

File: store/store.go
Comment (mardy) [reply store/store.go:748]: Then maybe even
```go
if err != nil || cloudInfo == nil {
    return "", err
}
```

**Pull request merged**
